src/components/dialogs/edit-package-details-dialog.tsx

```diff
@@ -0,0 +1,191 @@
+import {
+  Dialog,
+  DialogContent,
+  DialogHeader,
+  DialogTitle,
+  DialogDescription,
+} from "../ui/dialog"
+import { Button } from "../ui/button"
+import { useState, useEffect, useMemo } from "react"
+import { useMutation, useQueryClient } from "react-query"
+import { createUseDialog } from "./create-use-dialog"
+import { useAxios } from "@/hooks/use-axios"
+import { useToast } from "@/hooks/use-toast"
+import { Textarea } from "../ui/textarea"
+import { Input } from "../ui/input"
+import { Label } from "../ui/label"
+
+const isValidUrl = (url: string): boolean => {
+  if (!url) return true
+  try {
+    new URL(url)
+    return true
+  } catch (e) {
+    return false
+  }
+}
+
+export const EditPackageDetailsDialog = ({
+  open,
+  onOpenChange,
+  packageId,
+  currentDescription,
+  currentWebsite,
+  onUpdate,
+}: {
+  open: boolean
+  onOpenChange: (open: boolean) => void
+  packageId: string
+  currentDescription: string
+  currentWebsite: string
+  onUpdate?: (newDescription: string, newWebsite: string) => void
+}) => {
+  const [description, setDescription] = useState(currentDescription)
+  const [website, setWebsite] = useState(currentWebsite)
+  const [websiteError, setWebsiteError] = useState<string | null>(null)
+  const axios = useAxios()
+  const { toast } = useToast()
+  const qc = useQueryClient()
+
+  useEffect(() => {
+    if (open) {
+      setDescription(currentDescription)
+      setWebsite(currentWebsite)
+      setWebsiteError(null)
+    }
+  }, [open, currentDescription, currentWebsite])
+
+  useEffect(() => {
+    if (website && !isValidUrl(website)) {
+      setWebsiteError("Please enter a valid URL (e.g., https://tscircuit.com)")
+    } else {
+      setWebsiteError(null)
+    }
+  }, [website])
+
+  const hasChanges = useMemo(() => {
+    return description !== currentDescription || website !== currentWebsite
+  }, [description, website, currentDescription, currentWebsite])
+
+  const isFormValid = useMemo(() => {
+    return !websiteError
+  }, [websiteError])
+
+  const updatePackageDetailsMutation = useMutation({
```

Comment:
it's good to isolate heavy hooks like this into `lib/hooks` somewhere (can be refactored in future PR)

e.g. `usePackageDetailsMutation` (it's ok for it to be a thin wrapper over the `useMutation` params)